src/gluonts/dataset/artificial/_base.py

```diff
@@ -681,19 +688,23 @@ def trim_ts_item_end(x: DataEntry, length: int) -> DataEntry:
         the last prediction_length time points from the target and dynamic
         features."""
         y = dict(
-            item_id=x["item_id"],
-            start=x["start"],
-            target=x["target"][:-length],
+            item_id=x[FieldName.ITEM_ID],
+            start=x[FieldName.START],
+            target=x[FieldName.TARGET][:-length],
         )
 
-        if "feat_dynamic_cat" in x:
-            y["feat_dynamic_cat"] = x["feat_dynamic_cat"][:, :-length]
-        if "feat_dynamic_real" in x:
-            y["feat_dynamic_real"] = x["feat_dynamic_real"][:, :-length]
-        if "feat_dynamic_cat" in x:
-            y["feat_dynamic_cat"] = x["feat_dynamic_cat"]
-        if "feat_dynamic_real" in x:
-            y["feat_dynamic_real"] = x["feat_dynamic_real"]
+        if FieldName.FEAT_DYNAMIC_CAT in x:
+            y[FieldName.FEAT_DYNAMIC_CAT] = x[FieldName.FEAT_DYNAMIC_CAT][
+                :, :-length
+            ]
+        if FieldName.FEAT_DYNAMIC_REAL in x:
+            y[FieldName.FEAT_DYNAMIC_REAL] = x[FieldName.FEAT_DYNAMIC_REAL][
+                :, :-length
+            ]
+        if FieldName.FEAT_DYNAMIC_CAT in x:
+            y[FieldName.FEAT_DYNAMIC_CAT] = x[FieldName.FEAT_DYNAMIC_CAT]
+        if FieldName.FEAT_DYNAMIC_REAL in x:
+            y[FieldName.FEAT_DYNAMIC_REAL] = x[FieldName.FEAT_DYNAMIC_REAL]
```

Comment:
These last two `if`s seem to undo what the first two did. Are we sure this is right?

Reply:
yes, this looked strange to me as well.  Based on the definitions of these functions, I will remove the last two lines.

Reply:
@jgasthaus do you have any idea about this?